pkg/daemon/ceph/client/command.go

```diff
@@ -166,7 +166,12 @@ func (c *CephToolCommand) run() ([]byte, error) {
 	if command == RBDTool {
 		if c.RemoteExecution {
 			output, stderr, err = c.context.RemoteExecutor.ExecCommandInContainerWithFullOutputWithTimeout(ProxyAppLabel, CommandProxyInitContainerName, c.clusterInfo.Namespace, append([]string{command}, args...)...)
-			if stderr != "" || err != nil {
+			// err != nil and stderr != "" are both valid errors and may occur independently. Apply
+			// the error message or stderr as needed, then join after to create the returned err.
+			if stderr != "" {
+				logger.Warningf("message reported on stderr from %q command %v: %s", RBDTool, args, stderr)
```

Comment:
In the error case I was seeing, the warning looks like this now.

2022-01-25 19:01:32.856345 W | cephclient: message reported on stderr from "rbd" command [pool init device_health_metrics]: rbd: pool already registered to a different application.

Reply:
Shall we log this only if err == nil? It could be duplicate if there is an err and it's logged at a higher level.